packages/config-loader/src/lib/schema/filtering.ts

```diff
@@ -15,13 +15,44 @@
  */
 
 import { JsonObject, JsonValue } from '@backstage/types';
+import { has } from 'lodash';
 import {
   ConfigVisibility,
   DEFAULT_CONFIG_VISIBILITY,
   TransformFunc,
   ValidationError,
 } from './types';
 
+/**
+ * This filters data by deprecation status to only include those which have been deprecated
+ *
+ * @param data - configuration data
+ * @param deprecationBySchemaPath - mapping of schema path to deprecation description
+ * @returns deprecated configuration keys with their deprecation description
+ */
+export function filterByDeprecated(
```

Comment:
Since we're not actually filtering the configuration object by deprecation status, it might make more sense for me to move this elsewhere and rename it to better explain that it is mapping configuration data to deprecation status.

Reply:
I've moved it into `filterByVisibility`.

Currently, deprecations are added regardless of visibility. We could, however, change it to only include deprecations of a given visibility -- lmk.